system_metrics_collector/test/moving_average_statistics/test_moving_average_statistics.cpp

```diff
@@ -0,0 +1,194 @@
+  for (double d : TEST_DATA) {
+    moving_average_statistics->add_item(d);
+    ASSERT_EQ(++expected_count, moving_average_statistics->get_count());
+  }
+
+  const double expected_minimum = -3.5;
+  EXPECT_EQ(moving_average_statistics->min(), expected_minimum);
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_minimum_empty) {
+  ASSERT_TRUE(std::isnan(moving_average_statistics->min()));
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_standard_deviation) {
+  for (double d : TEST_DATA) {
+    moving_average_statistics->add_item(d);
+  }
+
+  EXPECT_DOUBLE_EQ(moving_average_statistics->standardDeviation(), EXPECTED_STD);
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_get_statistics) {
+  for (double d : TEST_DATA) {
+    moving_average_statistics->add_item(d);
+  }
+
+  auto result = moving_average_statistics->getStatistics();
+  EXPECT_DOUBLE_EQ(result.average, EXPECTED_AVG);
+  EXPECT_DOUBLE_EQ(result.min, EXPECTED_MIN);
+  EXPECT_DOUBLE_EQ(result.max, EXPECTED_MAX);
+  EXPECT_DOUBLE_EQ(result.standard_deviation, EXPECTED_STD);
+  EXPECT_DOUBLE_EQ(result.sample_count, EXPECTED_SIZE);
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_get_statistics_int) {
+  auto data_int = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+  const double expected_average = 5.5;
+  const double expected_minimum = 1;
+  const double expected_maximum = 10;
+  const double expected_std = 2.8722813232690143;
+  const int expected_size = 10;
+
+  for (int d : data_int) {
+    moving_average_statistics->add_item(d);
+  }
+
+  auto result = moving_average_statistics->getStatistics();
+  EXPECT_DOUBLE_EQ(result.average, expected_average);
+  EXPECT_DOUBLE_EQ(result.min, expected_minimum);
+  EXPECT_DOUBLE_EQ(result.max, expected_maximum);
+  EXPECT_DOUBLE_EQ(result.standard_deviation, expected_std);
+  EXPECT_DOUBLE_EQ(result.sample_count, expected_size);
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_standard_deviation_empty) {
+  ASSERT_TRUE(std::isnan(moving_average_statistics->standardDeviation()));
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_reset) {
+  moving_average_statistics->add_item(0.6);
+  moving_average_statistics->reset();
+  ASSERT_TRUE(std::isnan(moving_average_statistics->average()));
+  moving_average_statistics->add_item(1.5);
+  EXPECT_EQ(moving_average_statistics->average(), 1.5);
+}
+
+TEST_F(MovingAverageStatisticsTestFixture, test_thread_safe) {
+  std::atomic<int> total_sum(0);
+  std::atomic<int> count(0);
+
+  std::thread t1([this, &count, &total_sum]() {
+      for (double i = 1; i < 1101.0; i++) {
+        moving_average_statistics->add_item(i);
+        count++;
+        total_sum += i;
+      }
+    });
+  std::thread t2([this, &count, &total_sum]() {
+      for (double i = 1; i < 2101.0; i++) {
+        moving_average_statistics->add_item(i);
+        count++;
+        total_sum += i;
+      }
+    });
+  std::thread t3([this, &count, &total_sum]() {
+      for (double i = 1; i < 3101.0; i++) {
```

Comment:
it is preferable to index loops by int and cast to double when calling `add_item`